teps/0070-tekton-catalog-task-platform-support.md

```diff
@@ -0,0 +1,129 @@
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Catalog side](#catalog-side)
+  - [Hub side](#hub-side)
+  - [Tekton CLI side](#tekton-cli-side)
+- [References (optional)](#references-optional)
+
+## Summary
+
+This TEP proposes to add information about supported operating systems/hardware architectures (platforms) to the Tekton catalog resources and provide ability to the user to filter/search resources from the catalog for specific platform.
+
+## Motivation
+
+Tekton at this moment has releases for 4 hardware architectures(platforms) - `linux/amd64`, `linux/s390x`, `linux/ppc64le` and `linux/arm64`. Tekton catalog contains already great list of resources to run. However there is no information about which resource can be executed on which platform. As a result user can fail running a task on top of not supported hardware architecture.
+
+### Goals
+
+- Define the way to add supported platforms into resource definition in the Tekton catalog.
+- Define the ways to show information about platforms to the user.
+
+### Non-Goals
+
+- Define modifications in Tekton to better support running tasks on different platforms ([TEP-0041](https://github.com/tektoncd/community/pull/310)).
+- Limit/hide ("no-show" or "no-access") catalog resources based on host platform.
+
+### Use Cases (optional)
+
+- Filter the resources in the Hub UI using platforms.
+- Search the resources with `tkn hub` for specific platform(s).
+- Get information about supported platforms for specific resource in the Hub UI/`tkn hub`.
+
+## Requirements
+
+- Platform information in the catalog resource should be optional.
+- Information how to identify/prove supported platform for the resource should be documented in the Tekton catalog.
+- Tekton Hub should automatically get/update information about platforms of the resources.
+- Hub UI should show all platforms available in database to filter.
+
+## Proposal
+
+### Catalog side
+
+Add new `tekton.dev/platforms` annotation to the resource yaml file with list of platforms separated by comma, for example
+
+```
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: image-build
+    tekton.dev/platforms: linux/amd64,linux/s390x
+```
+
+The `tekton.dev/platforms` annotation should be optional. 
+Information about supported platforms also should be added to the `README.md` as separate `Platform` section with extra use case examples for specific platforms if it is required.
+For instance, another value for default image parameter in the task.
+
+The conditions to add platforms into `tekton.dev/platforms` should be written in Tekton catalog documentation.
+The levels of verification are:
+- all the container images, used in the resource, have corresponding platform support.
+- the resource tests were executed on corresponding platform once for conrete version.
```

Comment:
How will be implementing this on catalog

Reply:
How will be differentiating like which test suit to run on which platform

Reply:
There are already tests available for the current tasks in catalog,  For instance for buildah 0.2 the tests are located at https://github.com/tektoncd/catalog/tree/main/task/buildah/0.2/tests. So in case if tests should be executed, they are already available.